torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -77,22 +92,32 @@ Welford<T> welford_combine(const Welford<T> &a, const Welford<T> &b) {
   auto result = Welford<T>{
     a.mean + delta * wb_over_w,
     a.m2 + b.m2 + delta * delta * a.weight * wb_over_w,
-    new_weight
+    new_weight,
+    a.index + b.index,
   };
   return result;
 }
 
 template <typename T>
-Welford<T> welford_combine(const Welford<T> &acc, T data) {
+Welford<T> welford_combine(const Welford<T> &acc, T data, const WeightRecp<T>* w=nullptr) {
   // Add a single data point
+  int64_t index = acc.index + 1;
   auto delta = data - acc.mean;
   auto new_weight = acc.weight + T(1);
-  auto new_mean = acc.mean + delta / new_weight;
+  T new_mean;
+  if constexpr (!IsVecType<T>::value) {
+    new_mean = acc.mean + delta / new_weight;
+  } else {
+    new_mean = acc.mean +
+        ((w == nullptr) ? delta / new_weight
+                        : delta * w->weight_recps[acc.index]);
```

Comment:
`operator[]` doesn't check the out-of-boundary access but `at` does. What's the perf impact if we use `at` here?

Reply:
`at` will be slower.  Add `acc.index >= w->weight_recps.size()` to ensure that it does not exceed the bounds and does not affect performance